tests/test_main.py

```diff
@@ -1,6 +1,94 @@
-# TODO: add tests
-from app.main import hello_world
+import pytest
 
+import app.main
 
-def test_hello_world():
-    assert hello_world() == "Hello, world!"
+
+@pytest.mark.parametrize(
+    "variable_name",
+    [
+        "lucky_number",
+        "pi",
+        "one_is_a_prime_number",
+        "name",
+        "my_favourite_films",
+        "profile_info",
+        "marks",
+        "sorted_variables",
+    ],
+)
+def test_variables_should_be_defined(variable_name):
+    assert hasattr(
+        app.main, variable_name
+    ), f"Variable '{variable_name}' should be defined."
+
+
+@pytest.mark.parametrize(
+    "variable,value",
+    [
+        ("lucky_number", 777),
+        ("pi", 3.14),
+        ("one_is_a_prime_number", False),
+        ("name", "Richard"),
+        (
+            "my_favourite_films",
+            [
+                "The Shawshank Redemption",
+                "The Lord of the Rings: The Return of the King",
+                "Pulp Fiction",
+                "The Good, the Bad and the Ugly",
+                "The Matrix",
+            ],
+        ),
+        ("profile_info", ("michel", "michel@gmail.com", "12345678")),
+        ("marks", {"John": 4, "Sergio": 3}),
+    ],
+)
+def test_variables_values(variable, value):
+    assert (
+        getattr(app.main, variable) == value
+    ), f"Variable '{variable}' should be equal to {value}."
+
+
+@pytest.mark.parametrize(
+    "variable_name",
+    [
+        "lucky_number",
+        "pi",
+        "one_is_a_prime_number",
+        "name",
+        "my_favourite_films",
+        "profile_info",
+        "marks",
+    ],
+)
+def test_variables_should_be_added_to_sorted_variables(variable_name):
+    sorted_variables = getattr(app.main, "sorted_variables")
+    assert (
+        getattr(app.main, variable_name) in sorted_variables["mutable"]
+        or getattr(app.main, variable_name) in sorted_variables["immutable"]
+    ), f"Variable '{variable_name}' should be added to 'sorted_variables'"
+
+
+def is_immutable(obj):
+    if (
```

Comment:
I know, that students don't know this, but still, you'd better use it:
https://stackoverflow.com/questions/33311258/python-check-if-isinstance-any-type-in-list

Reply:
Thanks, applied :)